util/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/AttributesUtil.java

```diff
@@ -0,0 +1,237 @@
+package com.google.cloud.healthcare.imaging.dicomadapter;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.dcm4che3.data.Attributes;
+import org.dcm4che3.data.Sequence;
+import org.dcm4che3.data.Tag;
+import org.dcm4che3.data.VR;
+import org.dcm4che3.net.Status;
+import org.dcm4che3.net.service.DicomServiceException;
+import org.dcm4che3.util.TagUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class AttributesUtil {
+
+  private static final String PN_ALPHABETIC = "Alphabetic";
+  private static final String PN_IDEOGRAPHIC = "Ideographic";
+  private static final String PN_PHONETIC = "Phonetic";
+  private static final String PN_DELIMITER = "=";
+
+  public static String getTagValue(JSONObject json, String tag) throws JSONException {
+    JSONObject jsonTag = json.getJSONObject(tag);
+    JSONArray valueArray = jsonTag.getJSONArray("Value");
+    if (valueArray.length() != 1) {
+      throw new JSONException(
+          "Expected single value for tag: "
+              + tag
+              + " in JSON:\n"
+              + valueArray.toString());
+    }
+    return valueArray.getString(0);
+  }
+
+  public static String getTagValueOrNull(JSONObject json, String tag) {
+    if (!json.has(tag)) {
+      return null;
+    }
+    JSONObject jsonTag = json.getJSONObject(tag);
+    if (!jsonTag.has("Value")) {
+      return null;
+    }
+    JSONArray valueArray = jsonTag.getJSONArray("Value");
+    if (valueArray.length() != 1) {
+      return null;
+    }
+    return valueArray.getString(0);
+  }
+
+  /**
+   * Returns array of QIDO-RS paths (more than 1 if multiple ModalitiesInStudy are present)
+   *
+   * @param attrs dcm4che Attributes to convert
+   * @param includeFields additonal includeFields for QIDO-RS
+   */
+  public static String[] attributesToQidoPathArray(Attributes attrs, String... includeFields)
```

Comment:
I didn't see any callers of this method specify include fields. Or is this from a follow-up PR?

Reply:
Used in CFindService: line 100 .

Reply:
What I meant was that no caller specifies the optional includeFields parameter, so it's not used code. My preference is to only submit code which is used somewhere, but not a big deal in this case, I will merge this now so we can start with part 3.

Reply:
CMove uses attributesToQidoPath with includefields. attributesToQidoPathArray is not called with  includefields, but it makes sense to leave this parameter for symmetry, I think.